### PR TITLE
update project files

### PR DESCRIPTION
**Pull request by davidism**
One of my goals with Pallets-Eco is to standardize and document how our projects
are organized and maintained. Using a standard layout and tool config across all
projects makes it much easier for anyone familiar with one project to jump in to
help another, and makes it easier to write one set of maintainer documentation
for all.

This is a general update to more closely match how the Pallets projects are laid
out. We try to use a minimal set of common tools without much custom config.
I've pre-squashed the commits, see pallets/flask#5457 with each commit split out
in the PR.

This also adds a message to the readme to inform potential contributors about
Pallets-Eco and how they can get involved.

### CODE REVIEW

**Comment by davidism** [comment]
I'll be running `ruff check` and `ruff format` in the next PR. Then adding typing for mypy and pyright in the PR after that.

**Comment by davidism** [comment]
This adds our standard publish workflow. In the future, creating a "release version N" PR and pushing a tag will trigger the workflow, which requires a review from a maintainer before uploading to PyPI using trusted publishing.

**Comment by davidism** [comment]
I realize this is impossible to review, but it's the same as every other Pallets project. I'm happy to answer any questions about what we use.

**Comment by nickjj** [comment]
What do you think about keeping the original author's copyright line item in the LICENSE file?

The current commit replaces `Copyright (c) Rob Hudson and individual contributors` with `Copyright 2011 Pallets Community Ecosystem` which disassociates Rob from the project. Maybe you can add both copyrights on separate lines?

**Comment by davidism** [comment]
My understanding is that this describes the copyright holder, not the original author. Pallets (described as the Pallets Community Ecosystem here for pallets-eco) hold the copyright since the project was transferred to us. Rob was also not the original author of Flask-DebugToolbar, but of django-debug-toolbar which it was based on. I can find no clear or consistent guidance from reliable sources about how this works in legal practice, so my instinct is to make it reflect the current state of the project.

**Review by nickjj — APPROVED** [review]
Ok, I was just making sure we do due diligence on if it's ok to remove the original name. Otherwise, I glanced through the other changes and they make sense.

**Comment by davidism** [comment]
See #258 and #259 for formatting/linting and typing.

File: .github/pull_request_template.md
Comment (jeffwidman) [inline .github/pull_request_template.md:23]: We currently don't have a `CHANGES.rst` in this repo:
* https://github.com/pallets-eco/flask-debugtoolbar/pull/198#issuecomment-1844149055

File: .github/workflows/lock.yaml
Comment (jeffwidman) [inline .github/workflows/lock.yaml:5]: I'm personally 👎 down on this policy, especially for a low-volume repo like this one... I'd rather let folks be able to swing by later and say "for anyone googling, here's new/updated info"... I understand why some busy projects choose to do it, but the problem it solves doesn't exist in this project. 

Just my $0.02, not something I feel strongly about.

File: .github/workflows/publish.yaml
Comment (jeffwidman) [inline .github/workflows/publish.yaml:17]: Curious--is this what drove the use of `requirements` files?

Because they add some complexity here vs just doing `pip install -e .` or `pip install -e .[tests,docs]`

I'm actually surprised you didn't go that route with Flask...

File: .github/workflows/publish.yaml
Comment (jeffwidman) [inline .github/workflows/publish.yaml:50]: It's been a minute since I worked with the `gh release create` command, but I thought `--draft` didn't support using the "auto-generate release notes" button? Anyone know if that's true? Otherwise we'll just have to try it and see... or if someone wants to maintain / enforce a `CHANGES.rst` file, that's fine too.

File: .readthedocs.yaml
Comment (jeffwidman) [inline .readthedocs.yaml:3]: ? I intentionally used `ubuntu-lts-latest` to minimize maintainer effort later... the odds of our builds breaking are low, and the consequences of it breaking are small since it's quickly caught via the RTD CI check and then easy enough for us to temp pin it at that point... but my guess is it'll never break even as ubuntu releases get bumped by RTD.

Not a big deal, just mentioning it... for low-volume repos, I'm more of a fan of floating...

File: pyproject.toml
Comment (jeffwidman) [inline pyproject.toml:8]: This is def better than the previous matt good email since he's no longer an active maintainer, but FYI that this email doesn't go to any of the current active maintainers (me/Nick/Grey/Mac). 

So I assume this implicitly is you/the core pallets team stepping up to monitor this email?

File: pyproject.toml
Comment (jeffwidman) [inline pyproject.toml:10]: Probably about time we move this out of Beta status 😁

File: pyproject.toml
Comment (jeffwidman) [inline pyproject.toml:24]: I don't monitor this, I doubt any of the other current maintainers do... totally cool with leaving it in, just an FYI this redirects folks to something we don't really look at.

**Review by jeffwidman — COMMENTED** [review]
Thanks so much for handling all this!

Sorry I was late for the review, I started yesterday and forgot to hit publish until today... 

A few minor observations, nothing too critical, and most of it is just my opinions rather than facts. 😄

File: pyproject.toml
Comment (davidism) [reply pyproject.toml:24]: For better or worse, we're building the Pallets community in one place. The place for chat is our official Discord server. If I could make a new decision today it would probably be Zulip or Matrix, but that's not something I have the bandwidth to work on right now. Regardless, it would be somewhere, and I would hope that members of the pallets-eco community would congregate there.

Additionally, while I know Flask-DebugToolbar is maintained fairly actively, that's actually unusual for a project in Pallets-Eco right now. The goal is to make it easier for members of the community to join in as maintainers to _any_ of the pallets-eco projects. Part of that is having a single place to go for coordination.

File: pyproject.toml
Comment (davidism) [reply pyproject.toml:10]: Yes, we should also think about what 1.0 looks like.

File: pyproject.toml
Comment (davidism) [reply pyproject.toml:8]: Eventually, I may have a way to create `flask-debugtoolbar@palletsprojects.com` as a forwarding address. For now, it's one general address. In practice, I have had maybe one legitimate email versus all the spam that this enables, so I don't think people are missing out regardless.

File: .readthedocs.yaml
Comment (davidism) [reply .readthedocs.yaml:3]: For now I'm just standardizing across projects, and pinning seemed better than not. What's nice is that once we're all standardized, we can use a tool like https://github.com/asottile/all-repos to update everything at once.

File: .github/workflows/publish.yaml
Comment (davidism) [reply .github/workflows/publish.yaml:50]: I know we had a conversation about changes in releases vs a file before. I didn't realize GitHub had this limitation. Perhaps the changelog can be generated by editing the release after publishing it as well? It seems like this is more a bug to be fixed by GitHub if it's not possible right now.

File: .github/workflows/publish.yaml
Comment (davidism) [reply .github/workflows/publish.yaml:17]: I don't like the using publicly installable extras for development only environments. It's also not possible to pin extras autoamtaically. pip-compile allows pinning multiple environments as well as automatic updates with Dependabot.

File: pyproject.toml
Comment (jeffwidman) [reply pyproject.toml:24]: Sounds great! 

> while I know Flask-DebugToolbar is maintained fairly actively, that's actually unusual for a project in Pallets-Eco right now.

Amazing, because we considered it a project almost on life support. 😄 

> The goal is to make it easier for members of the community to join in as maintainers to any of the pallets-eco projects. Part of that is having a single place to go for coordination.

👍 for reducing friction for community to get involved. 😍

File: pyproject.toml
Comment (jeffwidman) [reply pyproject.toml:8]: My vote is let's just go with this for now, and if it starts to be a problem you can break out forwarding addresses. Like you say, probably no one will even bother to use it.

File: .readthedocs.yaml
Comment (jeffwidman) [reply .readthedocs.yaml:3]: Makes sense. I tend to forget about the broader pallets-eco...

File: .github/workflows/publish.yaml
Comment (jeffwidman) [reply .github/workflows/publish.yaml:50]: I remember the details now, and I suspect it's a non-issue. 

The problem arose when I was automating :dependabot: releases to RubyGems... we wanted it entirely automated as soon as someone merged a version bump PR, and the "autogenerate notes" wasn't available via an API call.

But IIRC the button was there... it just required a human.

And since then, they may have updated the API calls anyway, as that was a while ago.

Anyway, we're good for now, and can fix forward if we run into hiccups in the process later.

File: .github/workflows/lock.yaml
Comment (jeffwidman) [reply .github/workflows/lock.yaml:5]: That's reasonable. Long term, as we've discussed elsewhere, hopefully we can also distribute the work of maintaining across all these projects to dedicated sub-teams as it's unreasonable to expect you to keep up with everything... you're already one of the more productive OSS maintainers I know just with what you accomplish with the core Pallets stuff.

**Pull request merged**
